### PR TITLE
fix random mask creation in test_maskedtensor

### PR DESCRIPTION
**Pull request by pmeier**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96334
* #96333
* #96124
* #97010
* #96332
* __->__ #97017
* #96125
* #96331

cc @mruberry

### CODE REVIEW

File: test/test_maskedtensor.py [hunk 1]
```diff
@@ -64,13 +64,9 @@ def _compare_mts(mt1, mt2, rtol=1e-05, atol=1e-08):
     if not _tensors_match(a, b, exact=False, rtol=rtol, atol=atol):
         raise ValueError("The data in MaskedTensor mt1 and MaskedTensor mt2 do not match")
 
-def _make_tensor_mask(shape, device):
-    return make_tensor(
-        shape, device=device, dtype=torch.bool, low=0, high=1, requires_grad=False
```
Comment (pmeier) [inline test/test_maskedtensor.py:69]: It doesn't matter anymore, but `high=1` was wrong here, since the high value is exclusive.

File: test/test_maskedtensor.py [hunk 2]
```diff
@@ -64,13 +64,9 @@ def _compare_mts(mt1, mt2, rtol=1e-05, atol=1e-08):
...
-    )
 
 def _create_random_mask(shape, device):
-    return torch.randint(0, 2, shape, device=device).bool()
```
Comment (pmeier) [inline test/test_maskedtensor.py:73]: `make_tensor` uses `low=0, high=2` by default for `dtype=torch.bool` and thus we don't have to specify it.

File: test/test_maskedtensor.py (hunk 1 above)
Comment (pmeier) [reply test/test_maskedtensor.py:69]: Not sure I understand. `torch.bool` is hardcoded here. Are are you just saying that `high` is also exclusive for integral dtypes? If so, that is true, but not the whole picture. `high` in `make_tensor` is exclusive for all dtypes.

**Review by pearu — APPROVED** [review]
LGTM! Thanks, @pmeier!

File: test/test_maskedtensor.py (hunk 1 above)
Comment (pearu) [reply test/test_maskedtensor.py:69]: Nvm, we have had the same bug for integer dtypes in past. I was just surprised that masked tensors still had it. It basically meant that all related tests were using degenerate masks (all false).

Comment (pmeier) [reply test/test_maskedtensor.py:69]: > all related tests were using degenerate masks (all false).

No, the tests were fine. `torch.testing.make_tensor` ignored `low` and `high` for `torch.bool`. #96332 in this stack fixes that, which also surfaces this error.

**Review by mruberry — APPROVED** [review]
